File: algorithm/leetcode524.longest-word-in-dictionary-through-deleting.cpp
class Solution {
public:
    string findLongestWord(string s, vector<string>& d) {
    	string ans;
    	int i, j, k;
    	bool ok;
    	for(i = 0; i < d.size(); ++i)
    	{
    		j = k = 0;
    		for(; j <s.size() && k<d[i].size(); ++j)
    		{
    			if(s[j] == d[i][k])
    				k++;
    		}
    		if(k == d[i].size())
    		{
    			if(d[i].size() > ans.size())
    				ans = d[i];
    			else if(d[i].size() == ans.size() && d[i] < ans)
    				ans = d[i];
    		}
    	}
    	return ans;
    }
};

File: algorithm/Python3_version/leetcode524.longest-word-in-dictionary-through-deleting.py
class Solution:
    def findLongestWord(self, s: str, d: List[str]) -> str:
        ans = ""
        n = len(d)
        m = len(s)
        for i in range(n):
            j = k = 0
            while j < m and k < len(d[i]):
                if s[j]==d[i][k]:
                    k += 1
                j += 1
            if k==len(d[i]):
                if (len(d[i]) > len(ans)) or (len(d[i])==len(ans) and d[i] < ans):
                    ans = d[i]
        return ans